### DISCUSSION TITLE
Unable to Obtain Thermophysical Properties with Specific Input Values

### DISCUSSION THREAD

**Discussion by ironwolfpy** [CoolProp/CoolProp#2449] [Q&A] [answered]
I tried to obtain the thermophysical properties of a fluid using PropsSI with the following input values. The programming language is Fortran.
`h = PropsSI("HMASS"//C_NULL_CHAR, "T"//C_NULL_CHAR, 3.151d2, "P"//C_NULL_CHAR, 2.5382367d6, "R410A"//C_NULL_CHAR)
`
However, the enthalpy value h returns as Infinity. There are other such combinations as well. Why does this happen?

**Comment by ibell** [comment]
Make sure you check the error code first. Here is how that is done in the python wrapper: https://github.com/CoolProp/CoolProp/blob/master/wrappers/Python/CoolProp/CoolProp.pyx#L348 and https://github.com/CoolProp/CoolProp/blob/master/wrappers/Python/CoolProp/CoolProp.pyx#L402

**Reply by ironwolfpy** [reply]
Thank you for your reply. I tried the same thing in python and got the following error code
`File "CoolProp\\CoolProp.pyx", line 391, in CoolProp.CoolProp.PropsSI
  File "CoolProp\\CoolProp.pyx", line 471, in CoolProp.CoolProp.PropsSI
  File "CoolProp\\CoolProp.pyx", line 358, in CoolProp.CoolProp.__Props_err2
ValueError: Two-phase inputs not supported for pseudo-pure for now : PropsSI("D","P",2538236.7,"T",315.1,"R410A")`
Is the only way to avoid this is to move a little further away from the values around the two-phase region?

**Comment by ibell** [comment, marked as answer]
The pseudo-pure EOS is not to be used for two-phase states. The _reason_ is because the phase that is in equilibrium with the bulk phase changes for two-phase states for mixtures and such an effect is not captured by the pure-fluid EOS
